xdsl/pipeline.py

```diff
@@ -0,0 +1,11 @@
+from abc import abstractmethod
+from xdsl.ir import MLContext, Operation
+
+
+class OperationPass:
```

Comment:
What is the guarantee of an OperationPass? Is it executed on each Operation? AFAIU, the classic LLVM passes are Module/Function/Loop passes, all of which are run on individual Modules/Functions/Loops, making them inherently parallel. Now, we don't actually do parallelism at this point in Python land, but we do require that each pass is run on valid IR. This is something I don't quite get from this pass. What is it supposed to be applied on?

Reply:
A lot of legit concerns. My current answer would be "whatever guarantees our current anonymous functions have".
So, your main concern on this point is acting on a generic Operation rather than on the ModuleOp level? I can switch back to ModuleOp for the moment. I was simply looking forward to nested passes.

Reply:
Yes, I think making this a `ModulePass` is the way to go for the moment :)

We can discuuss nesting/composing in next PRs/discussions.

Reply:
I'm not sure what you mean by guarantees, do you mean what is expected by implementations of this class? Like leaving the IR in a verifiable state?

Reply:
Yes, things like "consume valid IR and produces valid IR".